packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2753,93 +2745,15 @@ function flushPassiveMountEffects(firstChild: Fiber): void {
     }
 
     if ((fiber.effectTag & Update) !== NoEffect) {
-      switch (fiber.tag) {
-        case FunctionComponent:
-        case ForwardRef:
-        case SimpleMemoComponent:
-        case Block: {
-          flushPassiveMountEffectsImpl(fiber);
-        }
-      }
+      setCurrentDebugFiberInDEV(fiber);
+      commitPassiveEffectOnFiber(fiber);
+      resetCurrentDebugFiberInDEV();
     }
 
     fiber = fiber.sibling;
   }
 }
 
-function flushPassiveMountEffectsImpl(fiber: Fiber): void {
-  setCurrentDebugFiberInDEV(fiber);
-
-  const updateQueue: FunctionComponentUpdateQueue | null = (fiber.updateQueue: any);
-  const lastEffect = updateQueue !== null ? updateQueue.lastEffect : null;
-  if (lastEffect !== null) {
-    const firstEffect = lastEffect.next;
-    let effect = firstEffect;
-    do {
-      const {next, tag} = effect;
-
-      if (
-        (tag & HookPassive) !== NoHookEffect &&
-        (tag & HookHasEffect) !== NoHookEffect
-      ) {
-        if (__DEV__) {
-          if (
-            enableProfilerTimer &&
-            enableProfilerCommitHooks &&
-            fiber.mode & ProfileMode
-          ) {
-            startPassiveEffectTimer();
-            invokeGuardedCallback(
-              null,
-              invokePassiveEffectCreate,
-              null,
-              effect,
-            );
-            recordPassiveEffectDuration(fiber);
-          } else {
-            invokeGuardedCallback(
-              null,
-              invokePassiveEffectCreate,
-              null,
-              effect,
-            );
-          }
-          if (hasCaughtError()) {
-            invariant(fiber !== null, 'Should be working on an effect.');
-            const error = clearCaughtError();
-            captureCommitPhaseError(fiber, error);
-          }
-        } else {
-          try {
-            const create = effect.create;
-            if (
-              enableProfilerTimer &&
-              enableProfilerCommitHooks &&
-              fiber.mode & ProfileMode
-            ) {
-              try {
-                startPassiveEffectTimer();
-                effect.destroy = create();
-              } finally {
-                recordPassiveEffectDuration(fiber);
-              }
-            } else {
-              effect.destroy = create();
-            }
-          } catch (error) {
-            invariant(fiber !== null, 'Should be working on an effect.');
-            captureCommitPhaseError(fiber, error);
-          }
-        }
-      }
-
-      effect = next;
-    } while (effect !== firstEffect);
-
-    resetCurrentDebugFiberInDEV();
```

Comment:
This was a bug. The reset is inside the if statement but the setCurrentDebugFiberInDEV is outside it. Meaning it wouldn't get reset.